core/chain_manager.go

```diff
@@ -487,15 +546,37 @@ const (
 	SideStatTy
 )
 
+/*
+// WriteHeader inserts a potentially new header into the database, updating the
+// head-header-hash index too if greater than the previous.
+func (self *ChainManager) WriteHeader(db common.Database, header *types.Header) error {
+	self.wg.Add(1)
+	defer self.wg.Done()
+
+	// Store the header into the database
+	if err := WriteHeader(self.chainDb, header); err != nil {
+		glog.Fatalf("failed to write header: %v", err)
+		return err
+	}
+	// If the header's higher than our previous head, update
+	if header.
+}*/
+
 // WriteBlock writes the block to the chain (or pending queue)
-func (self *ChainManager) WriteBlock(block *types.Block, queued bool) (status writeStatus, err error) {
+func (self *ChainManager) WriteBlock(block *types.Block) (status writeStatus, err error) {
 	self.wg.Add(1)
 	defer self.wg.Done()
 
+	// Calculate the total difficulty of the block
+	td := new(big.Int).Add(block.Difficulty(), self.GetTd(block.ParentHash()))
```

Comment:
The function was used by the miner too, so I left it public. Although all current code still checks for parent existence, I agree, that it's better to double check than to potentially crash. I've added an extra check after retrieving the parent TD that will return a `core.ParentError` if the paren't TD cannot be found.